### PR TITLE
Improve Azure function error messages in sample apps

### PR DESCRIPTION
**Pull request by TaoChenOSU**
### Motivation and Context

Currently the pop-up error messages from the sample apps do not correctly describe what actually went wrong. For example:
image

### Description

This PR adds details to the error messages. For example:
image
image

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: samples/apps/chat-summary-webapp-react/src/hooks/SemanticKernel.ts [hunk 1]
```diff
@@ -50,16 +50,25 @@ export class SemanticKernel {
...
+                throw response.statusText + " => " + await response.text();
+            }
 
-        return (await response.json()) as T;
+            return (await response.json()) as T;
+        } catch (e) {
+            var additional_error_msg = ''
+            if (e instanceof TypeError) {
+                // fetch() will reject with a TypeError when a network error is encountered.
+                additional_error_msg = '\n\nPlease check you have the function running and that it is accessible by the web app'
```
Comment (dluc) [inline samples/apps/chat-summary-webapp-react/src/hooks/SemanticKernel.ts:69]: nit: I would remove the word "web" considering that this function can power any sort of app

File: samples/apps/auth-api-webapp-react/src/hooks/SemanticKernel.ts [hunk 2]
```diff
@@ -50,16 +50,25 @@ export class SemanticKernel {
...
+                throw response.statusText + " => " + await response.text();
+            }
 
-        return (await response.json()) as T;
+            return (await response.json()) as T;
+        } catch (e) {
+            var additional_error_msg = ''
+            if (e instanceof TypeError) {
+                // fetch() will reject with a TypeError when a network error is encountered.
+                additional_error_msg = '\n\nPlease check you have the function running and that it is accessible by the web app'
```
Comment (dluc) [inline samples/apps/auth-api-webapp-react/src/hooks/SemanticKernel.ts:69]: "web": see other command

**Review by dluc — APPROVED** [review]
LGTM. @craigomatic wdyt?

File: samples/apps/chat-summary-webapp-react/src/hooks/SemanticKernel.ts (hunk 1 above)
Comment (TaoChenOSU) [reply samples/apps/chat-summary-webapp-react/src/hooks/SemanticKernel.ts]: Removed

File: samples/apps/auth-api-webapp-react/src/hooks/SemanticKernel.ts (hunk 2 above)
Comment (TaoChenOSU) [reply samples/apps/auth-api-webapp-react/src/hooks/SemanticKernel.ts]: Removed

**Review by craigomatic — APPROVED** [review]
Love it! 

We have had a lot of people hitting throttling issues and wondering what's up, this will be a big help :)

**Review by dluc — APPROVED** [review]

**Pull request merged**